File: ROS_Tuts/Tut1_ROS_Node_ws/src/first_node/scripts/ros_first_node.py
#! /usr/bin/env python3

import rospy

if __name__ == '__main__':
    #select Ros node name
    nodeName = "first_node_rospy"

    #Init Node with its name
    rospy.init_node(nodeName)

    #Send a Log msg in terminal
    rospy.loginfo(nodeName+" has started")

    #Select the rate of node execution
    rate = rospy.Rate(10)

    #loop while the node is not killed
    while not rospy.is_shutdown():
        #Send a Log msg in terminal
        rospy.loginfo(nodeName+" is running")
        rate.sleep()

File: ROS_Tuts/Tut1_ROS_Node_ws/src/first_node/src/ros_first_node.cpp
#include <ros/ros.h>
#include <string>
int main(int argc, char **argv)
{
    //select Ros node name
    const std::string nodeName("first_node_roscpp");

    //Init Node with its name
    ros::init(argc, argv, nodeName);

    //create Node Handle
    ros::NodeHandle nh;

    //Send a Log msg in terminal
    const std::string logMsg(nodeName+" has Started");
    ROS_INFO("%s",logMsg.c_str());

    //create runining msg
    const std::string logMsgRun(nodeName+" is running");

    //Select the rate of node execution
    ros::Rate rate(10);

    //loop while the node is not killed
    while(ros::ok())
    {
        //Send a Log msg in terminal
        ROS_INFO("%s",logMsgRun.c_str());
        rate.sleep();
    }
}